single_kernel_mongo/managers/mongos_operator.py

```diff
@@ -315,11 +321,20 @@ def stop_charm_services(self) -> None:
     @override
     def restart_charm_services(self, force: bool = False) -> None:
         """Restarts the charm with the new configuration."""
-        if not self.state.cluster.config_server_uri:
-            logger.error("Cannot start mongos without a config server db")
-            raise MissingConfigServerError()
-
-        self.mongos_config_manager.configure_and_restart(force=force)
+        try:
+            if not self.state.cluster.config_server_uri:
+                logger.error("Cannot start mongos without a config server db")
+                raise MissingConfigServerError()
+            self.mongos_config_manager.configure_and_restart(force=force)
+        except WorkloadServiceError as e:
+            logger.error("An exception occurred when starting mongod agent, error: %s.", str(e))
+            self.charm.status_handler.set_running_status(
+                MongosStatuses.MONGOS_NOT_STARTED.value,
+                scope=Scope.UNIT,
+                statuses_state=self.state.statuses,
+                component_name=self.name,
+            )
+            raise
```

Comment:
How is this error handled? 
when raising, the charm is in an error state, preventing charm statuses from being displayed